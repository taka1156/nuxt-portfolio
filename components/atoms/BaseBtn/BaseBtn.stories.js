import BaseBtn from './BaseBtn.vue';

export default {
  title: 'Atoms/BaseBtn',
  component: BaseBtn,
  argTypes: {
    btnClick: { action: 'clicked' }
  }
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { BaseBtn },
  template: '<base-btn @btn-click="btnClick">Button</base-btn>'
});

export const Default = Template.bind({});
