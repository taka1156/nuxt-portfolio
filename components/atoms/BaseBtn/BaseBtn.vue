<template>
  <button class="base-btn" :class="`base-btn--${useType}`" @click="btnClick">
    <!-- @slot ボタンラベル -->
    <slot />
  </button>
</template>

<script>
export default {
  name: 'BaseBtn',
  props: {
    useType: {
      type: String,
      default: 'none',
      validator: function (value) {
        return ['none', 'card'].indexOf(value) !== -1;
      },
    },
  },
  methods: {
    btnClick() {
      /**
       * ボタンクリックイベント
       * @event btnClick
       * @type {event}
       */
      this.$emit('btn-click');
    },
  },
};
</script>

<style scoped>
/* css reset */
button {
  margin: 0;
  padding: 0;
  border: none;
  outline: none;
  background: transparent;
  cursor: pointer;
}

/* css reset */

.base-btn--card {
  display: block;
  font-size: 1em;
  width: 75%;
  margin: 10px auto;
  padding: 10px;
  background-color: white;
  color: gray;
  border: 1px solid gray;
  border-radius: 5px;
}

.base-btn--card:active,
.base-btn--card:hover,
.base-btn--card:focus {
  background-color: gray;
  color: white;
}
</style>
