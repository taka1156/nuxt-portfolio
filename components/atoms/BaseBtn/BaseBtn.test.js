import { mount } from '@vue/test-utils';
import BaseBtn from './BaseBtn.vue';

describe('BaseBtn', () => {
  const dummyButtonText = 'ダミーボタンテキスト';
  const baseBtn = mount(BaseBtn, {
    slots: {
      default: dummyButtonText
    }
  });

  it('slotがDOMに反映されているか', () => {
    const btnTag = baseBtn.find('button');
    expect(btnTag.text()).toBe(dummyButtonText);
    // スナップショット
    expect(btnTag.html()).toMatchSnapshot();
  });

  it('btnClickを実行した時、`btn-click`のイベントが発生するか', () => {
    baseBtn.vm.btnClick();
    expect(baseBtn.emitted('btn-click')).toBeTruthy();
  });
});
