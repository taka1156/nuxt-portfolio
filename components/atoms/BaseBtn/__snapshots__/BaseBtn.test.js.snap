// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseBtn slotがDOMに反映されているか 1`] = `"<button class=\\"base-btn--extend\\">ダミーボタンテキスト</button>"`;
